client/my-sites/checklist/controller.jsx

```diff
@@ -5,15 +5,39 @@
  */
 import React from 'react';
 import { get } from 'lodash';
+import page from 'page';
+import { isEnabled } from 'config';
 
 /**
  * Internal Dependencies
  */
 
 import ChecklistMain from './main';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import {
+	canCurrentUserUseCustomerHome,
+	canCurrentUserUseChecklistMenu,
+} from 'state/sites/selectors';
 
 export function show( context, next ) {
 	const displayMode = get( context, 'query.d' );
 	context.primary = <ChecklistMain displayMode={ displayMode } />;
 	next();
 }
+
+export function maybeRedirect( context, next ) {
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	const slug = getSelectedSiteSlug( state );
+	if ( isEnabled( 'customer-home' ) && canCurrentUserUseCustomerHome( state, siteId ) ) {
+		page.redirect( `/home/${ slug }` );
```

Comment:
This ignores query params on the /checklist url. There are a few places where we append query params([#](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/index.jsx#L498) and [#](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/index.jsx#L543)) to the /checklist url.